WordPress/openverse
# Add new feature notice for dark mode

## Fixes



Fixes #4157 by @fcoveram

## Description



This PR
- adds a new `VFeatureNotice` component that can be used for notifying people of new features
- adds a component `VDarkModeFeatureNotice` that uses the above to notify people that we now have a dark mode
- adds a spinning-glowing effect to the `VSelectField` based on the `showNewHighlight` boolean prop
- adds some logic to hide the notice and glow once the user has interacted with the feature
- _does not_ add logic to remove the notice based on time because we can do that with cleanup in a subsequent deployment

## Testing Instructions


0. Check out the PR and run the dev server.
1. Enable the "dark_mode_ui_toggle" feature.
2. Navigate to the homepage. You should see the notice and the glow effect around theme selector.
3. Click on the "See more" button in the notice or the theme selector. The notice and glow will go away, forever (or until you clear cookies).

## Checklist


      PRs) or the media properties generator (`ov just catalog/generate-docs media-props`
      for the catalog or `ov just api/generate-docs` for the API) where applicable.

[best_practices]:
  https://git-scm.com/book/en/v2/Distributed-Git-Contributing-to-a-Project#_commit_guidelines

## Developer Certificate of Origin


<details>
<summary>Developer Certificate of Origin</summary>

```
Developer Certificate of Origin
Version 1.1

Copyright (C) 2004, 2006 The Linux Foundation and its contributors.
1 Letterman Drive
Suite D4700
San Francisco, CA, 94129

Everyone is permitted to copy and distribute verbatim copies of this
license document, but changing it is not allowed.


Developer's Certificate of Origin 1.1

By making a contribution to this project, I certify that:

(a) The contribution was created in whole or in part by me and I
    have the right to submit it under the open source license
    indicated in the file; or

(b) The contribution is based upon previous work that, to the best
    of my knowledge, is covered under an appropriate open source
    license and I have the right under that license to submit that
    work with modifications, whether created in whole or in part
    by me, under the same open source license (unless I am
    permitted to submit under a different license), as indicated
    in the file; or

(c) The contribution was provided directly to me by some other
    person who certified (a), (b) or (c) and I have not modified
    it.

(d) I understand and agree that this project and the contribution
    are public and that a record of the contribution (including all
    personal information I submit with it, including my sign-off) is
    maintained indefinitely and may be redistributed consistent with
    this project or the open source license(s) involved.
```

</details>
